src/swsssdk/dbconnector.py

```diff
@@ -27,91 +93,154 @@ def load_sonic_db_config(sonic_db_file_path=SONIC_DB_CONFIG_FILE):
                 logger.warning(msg)
                 sonic_db_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config', 'database_config.json')
             with open(sonic_db_file_path, "r") as read_file:
-                SonicDBConfig._sonic_db_config = json.load(read_file)
+                # The database_config.json is loaded with '' as key. This refers to the local namespace.
+                SonicDBConfig._sonic_db_config[''] = json.load(read_file)
         except (OSError, IOError):
             msg = "Could not open sonic database config file '{}'".format(sonic_db_file_path)
             logger.exception(msg)
             raise RuntimeError(msg)
         SonicDBConfig._sonic_db_config_init = True
 
     @staticmethod
-    def db_name_validation(db_name):
+    def namespace_validation(namespace):
+        # Check the namespace is valid.
+        if namespace is None:
+            msg = "invalid namespace name given as input"
+            logger.warning(msg)
+            raise RuntimeError(msg)
+
+        # Check if the global config is loaded entirely or for the namespace
+        if namespace != '' and SonicDBConfig._sonic_db_global_config_init == False:
+            msg = "Load the global DB config first using API load_sonic_global_db_config"
+            logger.warning(msg)
+            raise RuntimeError(msg)
+
+        if SonicDBConfig._sonic_db_config_init == False:
+            SonicDBConfig.load_sonic_db_config()
+
+        if namespace not in SonicDBConfig._sonic_db_config:
+            msg = "{} is not a valid namespace name in configuration file".format(namespace)
+            logger.warning(msg)
+            raise RuntimeError(msg)
+
+    @staticmethod
+    def db_name_validation(db_name, namespace=None):
+        if namespace is None:
+            namespace = ''
```

Comment:
instead of repeating this, can you define a EMPTY_NAMESPACE and use it in the program.

Reply:
Defined a new function and use it to check and sent if namespace is None.